src/bootstrap/native.rs

```diff
@@ -216,21 +216,26 @@ pub(crate) fn is_ci_llvm_available(config: &Config, asserts: bool) -> bool {
         }
     }
 
-    if CiEnv::is_ci() {
+    if is_ci_llvm_modified(config) {
+        eprintln!(
+            "Detected LLVM as non-available: running in CI and modified LLVM in this change"
+        );
+        return false;
+    }
+
+    true
```

Comment:
Definitly non-blocking, just personal hate with `return`.

```suggestion
    if is_ci_llvm_modified(config) {
        eprintln!(
            "Detected LLVM as non-available: running in CI and modified LLVM in this change"
        );
        false
    } else {
        true
    }
```

Reply:
I can change that if you want, but note that this is one block in a (short) series of early returns. That's a pattern where I think it makes sense not to directly chain `if-else`, especially since the first one is a more complex nested return.

Reply:
I would avoid nested code blocks as much as possible. I don't think having `else` helping the code look better and easier to read

Reply:
I've left it alone for now.